theano/tensor/nnet/abstract_conv.py

```diff
@@ -139,20 +143,17 @@ def conv2d_grad_wrt_inputs(output_grad,
         channels, input rows, input columns).  This is the tensor that
         will be upsampled or the output gradient of the convolution
         whose gradient will be taken with respect to the input of the
-        convolution.  See the optional parameter
-        ``output_grad_shape``.
+        convolution.
     filters : symbolic 4D tensor
         set of filters used in CNN layer of shape (output channels,
         input channels, filter rows, filter columns).  See the
         optional parameter ``filter_shape``.
-    output_grad_shape : list of 4 symbolic or real ints
-        The shape of the output_grad parameter.  Optional, possibly
-        used to choose an optimal implementation.  You can give
-        ``None`` for any element of the list to specify that this
-        element is not known at compile time.
-    input_shape : list of 2 symbolic or real ints
-        The shape (row and column size) of the input (upsampled)
-        parameter.  Not Optional, since given the output_grad_shape
+    input_shape : [None/int/Constant] * 2 + [Tensor/int/Constant] * 2
+        The shape of the input (upsampled) parameter.
+        A tuple/list of len 4, with the first two dimensions
+        being None or int or Constant and the last two dimensions being
+        Tensor or int or Constant.
+        Not Optional, since given the output_grad shape
         and the subsample values, multiple input_shape may be
         plausible.
     filter_shape : list of 4 symbolic or real ints
```

Comment:
Please modify the description of `filter_shape` to follow the previous convention i.e. use as type `[None/int/Const] * 4` and add in the description that it's a tuple/list of len 4.
+
Also please add to the `border_mode` docstring in the `half` description that it's also known as `same` elsewhere.